dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -688,7 +693,7 @@ private void addDockstoreYmlVersionToWorkflow(String repository, String gitRefer
      * @param yamlAuthors
      * @param version
      */
-    private void addDockstoreYmlAuthorsToVersion(final List<YamlAuthor> yamlAuthors, Version version) {
+    private void addDockstoreYmlAuthorsToVersion(final List<YamlAuthor> yamlAuthors, Version version, User user) {
```

Comment:
Apologies, I know I proposed on every github release event ...
But seeing it in code, I recall the ORCID api was less stable than the other APIs we deal with especially when we were testing against their sandbox, this makes me wonder about performance. In particular, I worry that checking on every event may have performance implications  

I wonder if this should have a way to back off and space things out a bit since we know beforehand that orcid info is probably not updated often anyway. 

Brainstorming, 
1) we change this to a batched process, for example we create a new endpoint to reach out to ORCID nightly to update OrcidAuthors  
2) we could add an explicit column to OrcidAuthor to record when we last checked and only update if we're a week out from that
3) we may be able to take advantage of the automatic update timestamp ... but would need to make sure the timestamp gets updated on a no-op (when the author info doesn't change) or just make sure to update something every time

Reply:
1. Wouldn't frozen/snapshotted versions complicate this? Or maybe there's a 4th case, where when the user snapshots, we block on harvesting ORCID authors if it hasn't been done already.
2. On the stability, it would take 30 seconds or something like that to time out? Could we set an explicit timeout? But even if we did that, it doesn't solve the broader issue.

Reply:
Yeah, I haven't thought about the implications for snapshotted versions. As it is right now with my PR, snapshotting a version would not freeze the ORCID author info. 

For example: Version A and B could list the same ORCID author in their .dockstore.yml. If Version A is snapshotted, a GitHub release of Version B will update the ORCID author that also belongs to Version A because ORCID authors are unique on their ORCID IDs in the DB.